### ISSUE TITLE
PHP warning on new page creation / save / edit

### ISSUE THREAD

**Issue by adriantoll** [10up/gutenbridge#107]
### Describe the bug

I'm seeing these warnings in my PHP logs on a completely vanilla WP installation and only Convert To Blocks installed:

```
PHP Warning:  Undefined array key "content" in public/wp-content/plugins/convert-to-blocks/includes/ConvertToBlocks/RESTSupport.php on line 91
PHP Warning:  Trying to access array offset on value of type null in public/wp-content/plugins/convert-to-blocks/includes/ConvertToBlocks/RESTSupport.php on line 91
```

### Steps to Reproduce

1. Start with a vanilla WP installation
2. Install and activate Convert To Blocks
3. Click Pages > Add new
4. Check PHP logs for the warning

This happens when saving the new page too, and on subsequent attempts to edit the page.

### Screenshots, screen recording, code snippet

Dumping the `$data` variable used on the line mentioned gives a full set of post data for the edited page, but then seems to be picking up and trying to evaluate an incomplete set of data from other pages, e.g.

```
[14-Mar-2023 18:57:34 UTC] Array
(
    [id] => 2
    [title] => Array
        (
            [raw] => Sample Page
            [rendered] => Sample Page
        )

    [parent] => 0
)
```

There's no `[content]` array key there, hence the warning.

### WordPress information

```
### wp-core ###

version: 6.1.1
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: undefined
https_status: true
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: production
user_count: 1
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /Users/[redacted]/Sites/wordpress-vanilla/public
wordpress_size: 49.46 MB (51866323 bytes)
uploads_path: /Users/[redacted]/Sites/wordpress-vanilla/public/wp-content/uploads
uploads_size: 5.83 MB (6113895 bytes)
themes_path: /Users/[redacted]/Sites/wordpress-vanilla/public/wp-content/themes
themes_size: 2.51 MB (2636074 bytes)
plugins_path: /Users/[redacted]/Sites/wordpress-vanilla/public/wp-content/plugins
plugins_size: 65.34 KB (66911 bytes)
database_size: 2.64 MB (2768896 bytes)
total_size: 60.51 MB (63452099 bytes)

### wp-active-theme ###

name: Twenty Twenty-Three (twentytwentythree)
version: 1.0
author: the WordPress team
author_website: https://wordpress.org
parent_theme: none
theme_features: core-block-patterns, post-thumbnails, responsive-embeds, editor-styles, html5, automatic-feed-links, block-templates, widgets-block-editor
theme_path: /Users/[redacted]/Sites/wordpress-vanilla/public/wp-content/themes/twentytwentythree
auto_update: Disabled

### wp-plugins-active (1) ###

Convert to Blocks: version: 1.1.1, author: 10up, Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1808
imagemagick_version: ImageMagick 7.1.0-61 Q16-HDRI aarch64 20866 https://imagemagick.org
imagick_version: 3.7.0
file_uploads: File uploads is turned off
post_max_size: 10G
upload_max_filesize: 10G
max_effective_size: 10 GB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 256 GB
	imagick::RESOURCETYPE_DISK: 9.2233720368548E+18
	imagick::RESOURCETYPE_FILE: 192
	imagick::RESOURCETYPE_MAP: 128 GB
	imagick::RESOURCETYPE_MEMORY: 64 GB
	imagick::RESOURCETYPE_THREAD: 1
imagemagick_file_formats: 3FR, 3G2, 3GP, AAI, AI, APNG, ART, ARW, ASHLAR, AVI, AVIF, AVS, BAYER, BAYERA, BGR, BGRA, BGRO, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CR3, CRW, CUBE, CUR, CUT, DATA, DCM, DCR, DCRAW, DCX, DDS, DFONT, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, EXR, FARBFELD, FAX, FF, FILE, FITS, FL32, FLV, FRACTAL, FTP, FTS, FTXT, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, HALD, HDR, HEIC, HEIF, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, JXL, K25, KDC, KERNEL, LABEL, M2V, M4V, MAC, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORA, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PHM, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, POCKETMOD, PPM, PS, PS2, PS3, PSB, PSD, PTIF, PWP, QOI, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGB565, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, STRIMG, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TM2, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WEBM, WEBP, WMV, WPG, X3F, XBM, XC, XCF, XPM, XPS, XV, YAML, YCbCr, YCbCrA, YUV
gd_version: 2.3.3
gd_formats: GIF, JPEG, PNG, WebP, BMP, XPM
ghostscript_version: 10.00.0

### wp-server ###

server_architecture: Darwin 22.3.0 arm64
httpd_software: Apache/2.4.55 (Unix) OpenSSL/1.1.1t PHP/8.0.27
php_version: 8.0.27 64bit
php_sapi: apache2handler
max_input_variables: 1000
time_limit: 30
memory_limit: 128M
admin_memory_limit: 256M
max_input_time: 60
upload_max_filesize: 10G
php_post_max_size: 10G
curl_version: 7.87.0 (SecureTransport) OpenSSL/1.1.1t
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: false

### wp-database ###

extension: mysqli
server_version: 5.7.41
client_version: mysqlnd 8.0.27
max_allowed_packet: 4194304
max_connections: 151

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /Users/[redacted]/Sites/wordpress-vanilla/public/wp-content
WP_PLUGIN_DIR: /Users/[redacted]/Sites/wordpress-vanilla/public/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: Undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by dkotter** [comment]
@adriantoll Thanks for the report. I wasn't able to reproduce myself though looking at the code, it would be easy enough for us to ensure this warning never happens. I would love to figure out in what scenarios this does happen though before putting in a fix.

Looking at the example response you shared, are those the only fields you get (`id`, `title`, `parent`)? If so, there's lots of other fields missing besides just `content`, like `date`, `guid`, `status`, etc. All of those fields should be there, as those are all core to the WordPress REST API (which shouldn't be impacted by this plugin).

What account level are you testing with? Administrator or something else?

**Comment by adriantoll** [comment]
Hi @dkotter I was scratching my head as well - yes, those are the only field returned and I was logged in as an admin. This started when I was working on another site that was showing those errors, and I tried to replicate on a completely vanilla WP installation. It seemed to, but given that there wasn't anything obvious at your end I just double-checked by completely wiping the files and DB, and starting from scratch again, and I can't replicate the error this time. I can't think why it wouldn't replicate now, but it seems likely that it's something systemic at my end - I only opened the issue because it seemed to replicate on a fresh install - so I'll close this issue and open a new one if I can replicate it again on a fresh install (or figure out why it's happening on the other site, which has a lot more moving parts).
